components/google_cloud/sources/bigquery-query-results/README.md

```diff
@@ -0,0 +1,122 @@
+# BigQuery Query Results Source
+
+## Introduction
+
+This source emits new events with the results of an arbitrary SQL query provided
+by the user. It achieves so by periodically executing such query.
+
+Please note that the event source will **blindly** execute the query in an
+agnostic manner. This means that if the result of the query contains
+previously-seen records, those records will be re-sent in the event payload. It
+is the user's responsibility to customize the SQL query so that the amount of
+emitted events is bounded.
+
+Users can also provide a field/column name to use for de-duplication purposes.
+Depending
+
+## Prerequisites
+
+To set up your BigQuery credentials, go to the [Pipedream
+Apps](https://pipedream.com/apps) page, and click on the **CONNECT AN APP**
+button. Look for the **Google Cloud** app, and click on it.
+
+The form contains a field called `key_json`, which should be filled with the
+contents of a JSON key file of a [GCP service
+account](https://cloud.google.com/iam/docs/creating-managing-service-account-keys).
+
+Please note that if the targeted data source from BigQuery requires special
+permissions, you must grant those permissions to the service account prior to
+creating this event source. For example, if the data source of the targeted
+BigQuery table is based from a Google Sheets spreadsheet, you must share that
+spreadsheet with the service account, and grant at least **View** access.
+
+After filling in every field required in the form, and click on **SAVE**.
+
+## Usage
+
+1. Visit the [source
+   page](https://pipedream.com/sources/new?key=google_cloud-bigquery-query-results)
+2. Select the BigQuery account to use (see [the previous
+   section](#prerequisites) for information on how to set it up)
+3. You'll be prompted to enter the time interval for the executions. The default
+   value is 15 minutes.
+4. Select the _event size_ (i.e. the amount of records to be sent in a single
+   event). Select 1 (which is also the default value) if you prefer to send one
+   event per new row
+5. Enter the _SQL query_ to be executed at each iteration/execution of this
+   event source
+6. Optionally, select the _de-duplication key_ to use for the de-duplication
+   logic (see [the following section](#de-duplication-details) section for more
+   information)
+
+## De-duplication Details
+
+If users provide a valid deduplication key during setup, the event source will
```

Comment:
The "If" here makes it seem like the dedupe key is optional, when it's required in the source. Is that intentional?